pkg/certificate/providers/config.go

```diff
@@ -313,18 +313,17 @@ func (c *Config) getHashiVaultOSMCertificateManager(options VaultOptions) (certi
 		return nil, nil, fmt.Errorf("error getting Vault Root Certificate, got: %w", err)
 	}
 
-	certManager, err := certificate.NewManager(vaultCert, vaultClient, c.cfg.GetServiceCertValidityPeriod(), c.msgBroker)
+	certManager, err := certificate.NewManager(vaultCert, vaultClient, c.cfg.GetServiceCertValidityPeriod(), c.msgBroker, make(chan struct{}))
```

Comment:
Same as before, creating the channel here makes it impossible to access later to stop cert rotation.

Overall, I think in each `main()` where we create a cert manager, `Start` should be called right after that and then `Start` should take the channel in as a parameter so `NewCertManager` doesn't need to accept one in cases cert rotation isn't needed.

To be overly specific, I mean something like this:
```diff
diff --git a/cmd/osm-bootstrap/osm-bootstrap.go b/cmd/osm-bootstrap/osm-bootstrap.go
index 509997780..e736e4ff1 100644
--- a/cmd/osm-bootstrap/osm-bootstrap.go
+++ b/cmd/osm-bootstrap/osm-bootstrap.go
@@ -9,6 +9,7 @@ import (
 	"flag"
 	"fmt"
 	"os"
+	"time"
 
 	"github.com/pkg/errors"
 	"github.com/spf13/pflag"
@@ -182,6 +183,8 @@ func main() {
 		events.GenericEventRecorder().FatalEvent(err, events.InvalidCertificateManager,
 			"Error initializing certificate manager of kind %s", certProviderKind)
 	}
+	// watch for certificate rotation
+	certManager.Start(5*time.Second, stop)
 
 	// Initialize the crd conversion webhook server to support the conversion of OSM's CRDs
 	crdConverterConfig.ListenPort = constants.CRDConversionWebhookPort
diff --git a/pkg/certificate/manager.go b/pkg/certificate/manager.go
index d936228f4..4a3a576b8 100644
--- a/pkg/certificate/manager.go
+++ b/pkg/certificate/manager.go
@@ -22,8 +22,7 @@ func NewManager(
 	ca *Certificate,
 	client client,
 	serviceCertValidityDuration time.Duration,
-	msgBroker *messaging.Broker,
-	certRotation <-chan struct{}) (*Manager, error) {
+	msgBroker *messaging.Broker) (*Manager, error) {
 	if ca == nil {
 		return nil, errNoIssuingCA
 	}
@@ -34,17 +33,16 @@ func NewManager(
 		client:                      client,
 		serviceCertValidityDuration: serviceCertValidityDuration,
 		msgBroker:                   msgBroker,
-		certRotation:                certRotation,
 	}
 	return m, nil
 }
 
 // Start takes an interval to check if the certificate
 // needs to be rotated
-func (m *Manager) Start(checkInterval time.Duration) {
+func (m *Manager) Start(checkInterval time.Duration, certRotation <-chan struct{}) {
 	// iterate over the list of certificates
 	// when a cert needs to be rotated - call RotateCertificate()
-	if m.certRotation == nil {
+	if certRotation == nil {
 		log.Error().Msgf("Cannot start certificate rotation, certRotation is nil")
 		return
 	}
@@ -52,7 +50,7 @@ func (m *Manager) Start(checkInterval time.Duration) {
 	go func() {
 		for {
 			select {
-			case <-m.certRotation:
+			case <-certRotation:
 				ticker.Stop()
 				return
 			case <-ticker.C:
diff --git a/pkg/certificate/providers/config.go b/pkg/certificate/providers/config.go
index ef5edf31d..4d807ac96 100644
--- a/pkg/certificate/providers/config.go
+++ b/pkg/certificate/providers/config.go
@@ -246,12 +246,10 @@ func (c *Config) getTresorOSMCertificateManager() (*certificate.Manager, debugge
 		return nil, nil, errors.Errorf("Failed to instantiate Tresor as a Certificate Manager")
 	}
 
-	tresorCertManager, err := certificate.NewManager(rootCert, tresorClient, c.cfg.GetServiceCertValidityPeriod(), c.msgBroker, make(chan struct{}))
+	tresorCertManager, err := certificate.NewManager(rootCert, tresorClient, c.cfg.GetServiceCertValidityPeriod(), c.msgBroker)
 	if err != nil {
 		return nil, nil, fmt.Errorf("error instantiating osm certificate.Manager for Tresor cert-manager : %w", err)
 	}
-	// watch for certificate rotation
-	tresorCertManager.Start(checkCertificateExpirationInterval)
 	return tresorCertManager, tresorCertManager, nil
 }
```